app/facades/trip_facade.rb

```diff
@@ -6,78 +6,54 @@ def initialize(trip_params)
     @destination = trip_params[:destination]
     @trip = Trip.create
 
+
     # Add SidekiqWorker job here to build POIs and legs of trip
     # The worker will need the trip id/token and Google directions payload
     # Otherwise it will need to make a second Google API request
+
+    # @matt -- I think this is the right syntax -- Not sure
+
+    TripWorker.build_trip(_directions.steps)
```

Comment:
I believe `.perform` is the key method to use.  It gives access to `.perform_async` - Essentially perform right away on a separate thread.  

I am concerned that Sidekiq is designed for short quick jobs so I believe it will time out part way through if we batch the whole job into one worker.  We may need to do something like this article explains making one "master" worker that spawns "smaller workers" in a loop.  

https://dev.to/jamby1100/coding-sidekiq-workers-the-right-way-4jij#2-dont-make-your-workers-too-big

Reply:
Good to know, I"ll work around this